moveit_core/collision_detection_fcl/src/collision_env_fcl.cpp

```diff
@@ -434,6 +434,36 @@ void CollisionEnvFCL::notifyObjectChange(const ObjectConstPtr& obj, World::Actio
     }
     cleanCollisionGeometryCache();
   }
+  else if (action == World::MOVE_SHAPE)
+  {
+    auto it = fcl_objs_.find(obj->id_);
+    if (it == fcl_objs_.end())
+    {
+      ROS_ERROR_NAMED(LOGNAME, "Cannot move shapes of unknown FCL object: '%s'", obj->id_.c_str());
+      return;
+    }
+
+    if (obj->global_shape_poses_.size() != it->second.collision_objects_.size())
+    {
+      ROS_ERROR_NAMED(LOGNAME,
+                      "Cannot move shapes, shape size mismatch between FCL object and world object: '%s'. Respectively "
+                      "%zu and %zu.",
+                      obj->id_.c_str(), it->second.collision_objects_.size(), it->second.collision_objects_.size());
+    }
+
+    for (std::size_t i = 0; i < it->second.collision_objects_.size(); ++i)
+    {
+      it->second.collision_objects_[i]->setTransform(transform2fcl(obj->global_shape_poses_[i]));
+
+      // compute AABB, order matters
+      it->second.collision_geometry_[i]->collision_geometry_->computeLocalAABB();
+      it->second.collision_objects_[i]->computeAABB();
+    }
+
+    // Reregister the FCL object into the broad phase collision manager
```

Comment:
Ideally the comment would explain why this needs to happen. Is there no API for moving objects in fcl? (I know it's not implemented for the manager actually used)

Reply:
I will try to see why this is needed (Actually I just repeated what was done [here](https://github.com/moveit/moveit/blob/master/moveit_core/collision_detection_fcl/src/collision_env_fcl.cpp#L366C23-L399)).

I can't find any API to move the object in FCL, only the `CollisionEnvFCL::updateFCLObject` which reconstruct's the object...

There is 3 options to update the broadphase collision manager:
```
    for (std::size_t i = 0; i < it->second.collision_objects_.size(); ++i)
    {
      it->second.collision_objects_[i]->setTransform(transform2fcl(obj->global_shape_poses_[i]));

      // compute AABB, order matters
      it->second.collision_geometry_[i]->collision_geometry_->computeLocalAABB();
      it->second.collision_objects_[i]->computeAABB();

      // Option #1 (18 usec)
      // manager_->update(it->second.collision_objects_[i].get());
    }
    // Option #2 (15 usec)
    // manager_->update();
    
    // Option #3 (7 usec)
    // it->second.unregisterFrom(manager_.get());
    // it->second.registerTo(manager_.get());
```
The fastest way seems to be the original fix I proposed.